### PR TITLE
#141404673 Display the game log, leaderboard and donations

### PR DESCRIPTION
**Pull request by fmustapha**
**What does this PR do?**

Merges Feature/141404673/game log branch to develop branch.

**Description of Task to be completed?**
As a registered user I want to view a "game log" of previously played games so that I can reference the friends played with, number of rounds and overall winner of the different games played.
GIVEN I am a registered user that has logged in
WHEN I click on the "dashboard" link from the main navigation
THEN I see a list of previously played games with details as given above

WHEN I click on the "dashboard" link from the main navigation
THEN I also see the leaderboard for all games played on the platform

WHEN I click on the "dashboard" link from the main navigation
THEN I also see a list of all donations made by the logged in user.

**How should this be manually tested?**
Login with your username and password
click on 'play with friends' or 'play with strangers'
click on 'dashboard' link on the main menu
view game logs, leaderboard and donations made by clicking on the corresponding tabs

**Any background context you want to provide?**

N/A

**What are the relevant pivotal tracker stories?**
Registered users should be able to view a "game log" of past games [#141404673]

### CODE REVIEW

File: app/controllers/game.js [hunk 1]
```diff
@@ -60,3 +60,22 @@ exports.updateRecord = (req, res) => {
       }
     });
 };
+
+exports.leaderboard = (req, res) => {
+  Game.find({}, (error, allGames) => {
+    if (error) {
```
Comment (sdowoade) [inline app/controllers/game.js:66]: Which error are we catching in this case?

File: public/js/directives.js [hunk 2]
```diff
@@ -146,4 +146,78 @@ angular.module('mean.directives', [])
         }
       });
     },
-  })]);
+  })])
+  .directive('leaderboard', ['$http', $http => ({
+    restrict: 'EA',
+    link: (scope) => {
+      const getLeaderboard = () => {
+        $http.get('/api/leaderboard').then((response) => {
```
Comment (sdowoade) [inline public/js/directives.js]: Let's move the HTTP calls to a service.

File: app/controllers/game.js (hunk 1 above)
Comment (tonymontaro) [reply app/controllers/game.js:66]: All fixed, thanks.

File: app/controllers/users.js [hunk 3]
```diff
@@ -459,3 +459,12 @@ exports.user = (req, res, next, id) => {
       next();
     });
 };
+
+exports.donations = (req, res) => {
+  User.find({}, (error, allDonations) => {
+    if (error) {
+      return res.status(404).send({ error });
```
Comment (sdowoade) [inline app/controllers/users.js]: Let's change this to 500 as well.

Comment (fmustapha) [reply app/controllers/users.js]: Okay done, thanks.

**Review by sdowoade — APPROVED** [review]

**Pull request merged**